### PR TITLE
Update Parental Bond calculation

### PR DESCRIPTION
**Pull request by augmt**
I've updated Parental Bond according to what it says here: http://nuggetbridge.com/forums/topic/11282-how-to-correctly-compute-parental-bond-damage/

### CODE REVIEW

**Comment by Tapin42** [comment]
Thanks!

One question: Since you're uniq'ing the output array, doesn't that mess with xHKO%?  Overrepresenting the extreme values, in particular?  (Compared to http://nuggetbridge.com/forums/topic/11282-how-to-correctly-compute-parental-bond-damage/?p=157669)

**Comment by augmt** [comment]
Ah, you're right. Do you think that sorting the output array would be the way to go, then?

**Filtering the output array**
108+ Atk Parental Bond Mega Kangaskhan Sucker Punch vs. 252 HP / 0 Def Mega Abomasnow: 64-76 (32.4 - 38.5%) -- 98.4% chance to 3HKO
(64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76)

vs.

**Sorting the output array w/ .sort()**
108+ Atk Parental Bond Mega Kangaskhan Sucker Punch vs. 252 HP / 0 Def Mega Abomasnow: 64-76 (32.4 - 38.5%) -- 99.8% chance to 3HKO
(Trust me, it's an array with 256 values)

**Comment by Tapin42** [comment]
I haven't dug into the actual guts of the damage calc for fear of breaking something, but sorting it makes conceptual sense to me.  I would imagine displaying all 256 values to the user would be a pretty ugly UI -- if that's the UI you put in there "(Trust me..." it seems good to me ;-)

**Comment by augmt** [comment]
Don't worry about breaking anything.

I've made it so no text is shown under the mainResult span if the output array's length is 256, but you can have it say whatever you'd like.

Also the child's attack is boosted by 1 if the parent uses Power-Up Punch. It's not the definitive patch for Power-Up Punch, but it's a start, I guess.

File: damage.js [hunk 1]
```diff
@@ -534,14 +535,21 @@ function getDamageResult(attacker, defender, move, field) {
         }
         damage[i] = Math.max(1, damage[i]);
         damage[i] = pokeRound(damage[i] * finalMod / 0x1000);
-
-        // is 2nd hit half BP? half attack? half damage range? keeping it as a flat 1.5x until I know the specifics
         if (attacker.ability === "Parental Bond" && move.hits === 1 && (field.format === "Singles" || !move.isSpread)) {
-            damage[i] = Math.floor(damage[i] * 3/2);
+            child = $.extend(true, {}, attacker, {ability: ""});
+            if (move.name === 'Power-Up Punch') {
```
Comment (Tapin42) [inline damage.js:540]: Good catch, I hadn't even considered this.

**Comment by Tapin42** [comment]
It looks good to me.  Thanks again for doing this!  (and of course thanks a ton for all the other work you did to get it to this point!)

**Comment by Tapin42** [comment]
Merged, and will close #15 accordingly.

**Pull request merged**